Zenodo.js

```diff
@@ -114,38 +116,79 @@ function doWeb(doc, url) {
-	var type = detectWeb(doc, url);
-	var schemaType = ZU.xpathText(doc, '//div[@id="wrap"]//div[@itemscope]/@itemtype');
+	var abstract = ZU.xpathText(doc, '//meta[@name="description"]/@content');
+	var doi = ZU.xpathText(doc, '//meta[@name="citation_doi"]/@content');
+	var schemaType = ZU.xpathText(doc, '//div[contains(@class, "record-detail")]//div[@itemscope]/@itemtype');
+	var pdfURL = ZU.xpathText(doc, '//meta[@name="citation_pdf_url"]/@content');
+	var tags = ZU.xpath(doc, '//meta[@name="citation_keywords"]')
+	var cslURL = url.replace(/#.+/, "").replace(/\?.+/, "").replace(/\/export\/.+/, "") + "/export/csl"
+	//Z.debug(cslURL)
+	//	Z.debug(schemaType)
 	// Embedded Metadata
-	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
-	translator.setDocument(doc);
-
-	translator.setHandler('itemDone', function (obj, item) {
-		// sometimes DOIs have to been cleaned up
-		// e.g. https://zenodo.org/record/54747
-		var dois = item.DOI.split(", ");
-		var cleaned = [];
-		for (var i=0; i<dois.length; i++) {
-			if (dois[i].substr(0,3) == "10.") {
-				cleaned.push(dois[i]);
+	// use CSL JSON translator
+	ZU.processDocuments(cslURL, function(newDoc){
+		var text = ZU.xpathText(newDoc, '//div[@itemscope]//h3/following-sibling::pre');
+		//Z.debug(text)
+		var type = text.match(/"type": "(.+?)"/)[1]
+		text = text.replace(/publisher_place/, "publisher-place");
+		text = text.replace(/container_title/, "container-title");
+	
+		var trans = Zotero.loadTranslator('import');
+		trans.setTranslator('bc03b4fe-436d-4a1f-ba59-de4d2d7a63f7');
+		trans.setString(text);
+		trans.setHandler("itemDone", function(obj, item) {
+			if (!item.DOI) {
+				if (item.extra) {
+					item.extra += "\nDOI: " + doi;
+				}
+				else {
+					item.extra = "DOI: " + doi;
+				}
 			}
-		}
-		item.DOI = cleaned.join();
-		if (type != "journalArticle" && type != "conferencePaper") {
-			item.extra = "DOI: " + item.DOI;
-		}
-		if (schemaType == "http://schema.org/Dataset") {
-			item.extra = "itemType: dataset\n" + "DOI: " + item.DOI;
-		}
-		item.complete();
-	});
-
-	translator.getTranslatorObject(function(trans) {
-		trans.itemType = type;
-		trans.doWeb(doc, url);
+			//workaround while we don't have proper item type for data
+			if (schemaType.indexOf("Dataset") != -1) {
+				if (item.extra) {
+					item.extra += "\ntype: dataset";
+				}
+				else {
+					item.extra = "type: dataset";
+				}
+			}
+			
+			//get PDF attachment, otherwise just snapshot.
+			if (pdfURL) {
+				item.attachments.push({url:pdfURL, title: "Zenodo Full Text PDF", mimeType: "application/pdf"})
+			}
+			else {
+				item.attachments.push({url:url, title: "Zenodo Snapshot", mimeType: "text/html"})
+			}
+			for (var i = 0; i<tags.length; i++) {
+				item.tags.push(tags[i].content);
+			}
+			
+			//something is odd with zenodo's author parsing; fix it
+			for (var i = 0; i< item.creators.length; i++) {
+				if (!item.creators[i].firstName && item.creators[i].lastName.indexOf(",")!=-1) {
```

Comment:
https://www.zenodo.org/record/569304/export/csl#.WUgtNsaQy70
but it's also in one of the tests. I don't understand the problem -- have double-checked data entry for this one and we did that exactly as recommended.